MovieBrowser/Controller/ViewController.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  ViewController.swift
+//  MovieBrowser
+//
+//  Created by Abdellah LHASSANI on 3/1/18.
+//  Copyright © 2018 Abdellah LHASSANI. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+import AlamofireImage
+
+class ViewController: UITableViewController {
+
+    var movies : [Movie] = []
+    var selectedMovie: Movie?
+    var favoriteMovies : [Movie] = []
+    var defaults = UserDefaults.standard
+    var moviesApi = "/3/discover/movie?api_key=98ddab1a678013f4f420946ce3d8b605&language=en-US&sort_by=popularity.desc&page=1"
+    var genresApi = "/3/genre/movie/list?api_key=98ddab1a678013f4f420946ce3d8b605&language=en-US"
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        getMovies()
+        setFavoritesFromFileSystem()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+       super.viewDidAppear(animated)
+       setFavoritesFromFileSystem()
+        
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! MovieTableViewCell
+        cell.moviePosterImageView.layer.cornerRadius = 10.0
+        cell.moviePosterImageView.clipsToBounds = true
+        cell.movieTypeAndLengthLabel.adjustsFontSizeToFitWidth = true
+        cell.movieTypeAndLengthLabel.text = ""
+        if  let id = movies[indexPath.row].id, let votes = movies[indexPath.row].voteAverage,let title = movies[indexPath.row].title,let imageUrl = movies[indexPath.row].posterPath ,  let genres = movies[indexPath.row].genre{
```

Comment:
Extract movie from array for simplicity
use `let movie = movies[indexPath.row] `before if statement